x-pack/metricbeat/module/enterprisesearch/health/health.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package health
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/helper"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/mb/parse"
+)
+
+const (
+	// defaultScheme is the default scheme to use when it is not specified in
+	// the host config.
+	defaultScheme = "http"
+
+	// defaultPath is the default path to the mod_status endpoint on the
+	// Apache HTTPD server.
+	defaultPath = "/api/ent/v1/internal/health"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("enterprisesearch", "health", New,
+		mb.WithHostParser(hostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	http *helper.HTTP
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The Enterprise Search health metricset is currently in beta.")
+
+	http, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
+	}
+	return &MetricSet{
+		base,
+		http,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
```

Comment:
If possible, improve the description of this metricset like "Fetch makes GET `defaultPath` call to ESS to retrieve the metrics required".

In contrast to those metricsets that "Fetches `/` to get some info, then `/stats` to get more info and finally `/metrics` to generate an event" Funny enough this was a problem as late as today :smile: https://github.com/elastic/beats/pull/28162

Reply:
Done!